worker/manager/src/mirrors_qa_manager/docker.py

```diff
@@ -0,0 +1,141 @@
+# pyright: strict, reportMissingTypeStubs=false, reportUnknownMemberType=false, reportOptionalSubscript=false, reportUnknownVariableType=false, reportUnknownArgumentType=false
+import time
+from collections.abc import Callable
+from functools import wraps
+from pathlib import Path
+from typing import Any
+
+from docker import DockerClient
+from docker.errors import APIError, ImageNotFound, NotFound
+from docker.models.containers import Container, ExecResult
+from docker.models.images import Image
+
+from mirrors_qa_manager import logger
+from mirrors_qa_manager.settings import Settings
+
+
+def retry(
+    retries: int = Settings.DOCKER_API_RETRIES,
+    interval: int = Settings.DOCKER_API_RETRY_SECONDS,
+):
+    """Retry API calls to the Docker daemon on failure."""
+
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(func)
+        def wrapped(*args: Any, **kwargs: Any):
+            attempt = 0
+            while True:
+                attempt += 1
+                try:
+                    return func(*args, **kwargs)
+                # most 404 errors would be from incorrect image/tag name but
+                # we still want to avoid crashing on 404 due to temporary
+                # docker-hub issues
+                except (ImageNotFound, APIError) as exc:
+                    if attempt <= retries:
+                        logger.error(
+                            f"docker api error for {func.__name__} "
+                            f"(attempt {attempt}): {exc}"
+                        )
+                        time.sleep(interval * attempt)
+                        continue
+                    raise exc
+
+        return wrapped
+
+    return decorator
+
+
+def get_running_container_name() -> str:
+    """Determine the name of this container."""
+    with open("/etc/hostname") as fh:
```

Comment:
```suggestion
    return Path("/etc/hostname").read_text()
```

Looks like this zimfarm code expected the hotsname to be path-like (with a `/` then. Not sure if this is valid. Should be checked